openedx/core/djangoapps/content/course_overviews/tasks.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from celery import task
+from celery_utils.logged_task import LoggedTask
+from celery_utils.persist_on_failure import PersistOnFailureTask
+from django.conf import settings
+
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+log = logging.getLogger(__name__)
+
+
+DEFAULT_CHUNK_SIZE = 50
+
+
+def chunks(sequence, chunk_size):
+    return (sequence[index: index + chunk_size] for index in xrange(0, len(sequence), chunk_size))
+
+
+class _BaseTask(PersistOnFailureTask, LoggedTask):  # pylint: disable=abstract-method
+    """
+    Include persistence features, as well as logging of task invocation.
+    """
+    abstract = True
+
+
+def generate_course_overview(course_ids, all_courses=False, force_update=False, chunk_size=DEFAULT_CHUNK_SIZE):
+    """
+    Invokes asynchronous celery tasks to load course overviews for all courses specified in course_ids.
+    If all_courses is True, loads a course overview for every course in the modulestore.
+    """
+    if all_courses:
+        course_keys = [course.id for course in modulestore().get_course_summaries()]
+    else:
+        course_keys = [CourseKey.from_string(id) for id in course_ids]
```

Comment:
What is the behavior when an invalid course-key is specified?  Does the management command gracefully fail with information about which course-key was invalid?

Reply:
If fails ungracefully, but with information about which course-key was invalid, via an `InvalidKeyError`.  I can promote that to a CommandError instead.